src/flint/test/test_all.py

```diff
@@ -2930,14 +2962,18 @@ def quick_poly():
 
         assert +quick_poly() \
             == quick_poly()
+
         assert -quick_poly() \
-            == mpoly({(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4})
+            == mpoly(
+                {(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4} if P is not flint.nmod_mpoly
+                else {k: ctx.modulus() + v for k, v in {(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4}.items()}
+        )
 
         assert quick_poly() \
             + mpoly({(0, 0): 5, (0, 1): 6, (1, 0): 7, (2, 2): 8}) \
             == mpoly({(0, 0): 6, (0, 1): 8, (1, 0): 10, (2, 2): 12})
 
-        for T in [int, S, flint.fmpz, lambda x: P(x, ctx=ctx)]:
+        for T in [int, S, lambda x: P(x, ctx=ctx)]:
             p = quick_poly()
             p += T(1)
```

Comment:
Does `fmpz` not always work here?
```python
In [15]: ctx.from_dict({(1, 2): 3}) * fmpz(3)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[15], line 1
----> 1 ctx.from_dict({(1, 2): 3}) * fmpz(3)

TypeError: unsupported operand type(s) for *: 'flint.types.nmod_mpoly.nmod_mpoly' and 'flint.types.fmpz.fmpz'
```
If the exception is only for `nmod_mpoly` then I think it would be better to write that explicitly in the test:
```python
        for T in [int, S, lambda x: P(x, ctx=ctx)]:
            p = quick_poly()
            if type(p) is not nmod_mpoly:
                ...
```
Really though I think that we want `fmpz` and `int` to be compatible with all types (see also gh-178):
```python
In [16]: nmod_poly([1, 2], 3) * fmpz(2)
Out[16]: x + 2

In [17]: nmod_poly([1, 2], 3) + fmpz(2)
Out[17]: 2*x

In [15]: ctx.from_dict({(1, 2): 3}) * fmpz(3)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[15], line 1
----> 1 ctx.from_dict({(1, 2): 3}) * fmpz(3)

TypeError: unsupported operand type(s) for *: 'flint.types.nmod_mpoly.nmod_mpoly' and 'flint.types.fmpz.fmpz'
```

Reply:
It should, just an oversight. I'm adding `fmpz` support now, commits will come tomorrow evening